fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -317,18 +304,15 @@ def sync_buffers(self, blocking: bool = False) -> None:
             blocking (bool): wait for the operation to conclude.
         """
 
-        work_handles = []
+        futures = []
 
         for buffer in self.module.buffers(recurse=True):
-            work_handles.append(
```

Comment:
basically the crux of the issue is that, the handles are not a good sync primitive, they are just not correct, and the cuda events are not accessible because the stream being used is not accessible